### PR TITLE
Add minimal glossary

### PR DESCRIPTION
**Pull request by cbkerr**
## Description

My goal is to define the terms for signac-core "good enough" to get started to keep this moving.

I think the new hover extension is worth it to allow people to see the definition when hovering.

## Motivation and Context

#59

Work continues at https://docs.google.com/document/d/1_merhcK3ohas4IloE616yL7gypMRFcaQLh2oChExC7o

## Checklist:

- [x] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/signac-docs/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/signac-docs/blob/master/ContributorAgreement.md).
- [x] My code follows the [code style guideline](https://github.com/glotzerlab/signac-docs/blob/master/CONTRIBUTING.md#code-style) of this project.

### CODE REVIEW

File: docs/source/glossary.rst [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+.. glossary::
+
+   parameter
+      A variable, like `T` or `version` or `bench_number`. The smallest unit in **signac**. Specifically, these are the dictionary keys of the state point.
+
+   state point
+      A dictionary of parameters and their values that uniquely specifies a :term:`job`, kept in sync with the file `signac_statepoint.json`.
+
+   job
+      An object holding data and metadata of a :term:`state point`.
```
Comment (b-butler) [inline docs/source/glossary.rst:13]:
```suggestion
      An object uniquely identified by a :term:`state point` containing metadata and data associated with the :term:`state point`.
```

File: docs/source/glossary.rst [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+   job directory
+      The directory, named for the :term:`job id`, created when a job is initialized containing all data and metadata pertaining to the given job.
+
+   job id
+      A unique MD-5 hash of a job's state point that is used to identify a job.
```
Comment (b-butler) [inline docs/source/glossary.rst:19]:
```suggestion
   job id
      A MD-5 hash of a job's state point that is used to distinguish between jobs.

   job directory
      The directory, named for the :term:`job id`, created when a job is initialized containing all data and metadata pertaining to the given job.
```

File: docs/source/glossary.rst [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+   job document
+      A persistent dictionary for storage of simple key-value pairs in a job, kept in sync with the file `signac_job_document.json`.
+
+   project
+      The primary interface to access and work with jobs and their data.
```
Comment (b-butler) [inline docs/source/glossary.rst:25]:
```suggestion
   project
      The full collection of associated jobs. The primary interface to access and work with jobs and their data.
```

File: docs/source/glossary.rst [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+   workspace
+      The directory that contains all job directories of a **signac** project.
+
+   project schema
+      The emergent database schema as defined by jobs in the project workspace. The set of all keys present in all state points, as well as their range of values.
+
+   signac schema
+      a database schema that defines how signac reads configuration options, currently on v2.
```
Comment (b-butler) [inline docs/source/glossary.rst:34]:
```suggestion
      a configuration schema that defines accepted options and values, currently on v2.
```

File: docs/source/jobs.rst [hunk 5]
```diff
@@ -23,7 +23,7 @@ In other words, all data associated with a particular job should be a direct or
 
 .. important::
 
-    Every parameter that, when changed, would invalidate the job's data, should be part of the *state point*; all others should not.
+    Every :term:`parameter` that, when changed, would invalidate the job's data, should be part of the *state point*; all others should not.
```
Comment (b-butler) [inline docs/source/jobs.rst:26]:
```suggestion
    Every :term:`parameter` that, when changed, would invalidate the job's data, should be part of the :term:`state point`; all others should not.
```

**Review by b-butler — COMMENTED** [review]
Is this meant to cover all uses of the glossary terms?

**Comment by cbkerr** [comment]
> Is this meant to cover all uses of the glossary terms?

Yes, these terms shouldn't have multiple definitions.

I tried to pick the ones that had most settled in the [google doc](https://docs.google.com/document/d/1_merhcK3ohas4IloE616yL7gypMRFcaQLh2oChExC7o/).

**Comment by b-butler** [comment]
I more meant, that many of the terms do not use hoverxref in the rest of the documentation.

**Review by b-butler — APPROVED** [review]

**Pull request merged**